### DISCUSSION TITLE
Pass triggering user information to DAG

### DISCUSSION THREAD

**Discussion by yupbank** [apache/airflow#24812] [Ideas]
**Description**

It would be very useful if we can pass triggering user information to DAG. So that we can bring over the info to the downstream system for security auditing purpose 
**Use case / motivation**

Expose some macro in the DAG, so user can use the trigger user info at will

**Are you willing to submit a PR?**

If someone can point me in the right direction 
**Related Issues**

No

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by eladkal** [comment]
In manual jobs there is a specific user who triggered it - that doesn't exist for scheduled jobs.
What value are you expecting to have for scheduled jobs?

**Comment by yupbank** [comment]
I'm looking for the manual jobs actually, currently, we can pass the info through the `dag_run.conf` in `cli` use case. 
but the web UI `dag_run.conf` is a bit complicated. I'm wondering if it makes sense to add some `current_user` field in the default conf https://github.com/apache/airflow/blob/main/airflow/www/views.py#L1490 here
something like `default_conf=json.dumps(dict(current_user=current_user.user.username))` 

it would be amazing if a default default_conf can be configured for all CLI and web UI submissions. Since airflow would perform checks on the triggering user anyway. why not pass it in to the DAG so default_conf actually have some default info

**Comment by yupbank** [comment]
e.g., i could define a BashOperator that login to a web server, in which all the credentials are managed by connections for sure. but it would also be nice to tell the web server that who triggered the job manually if that happened

**Comment by uranusjr** [comment]
Not commenting on whether this is a good idea or not, IMO a scheduled job could be categorised under either the DAG owner (since they scheduled the DAG) or `None`.

**Comment by yupbank** [comment]
maybe let me re-state the feature: allow the user to customize the default `DAG.default_conf` in the web console?

**Comment by santosh-d3vpl3x** [comment]
`airflow.models.log.Log` should already contain the information about trigger. Here the owner field indicates user performing actions.

You can listen on `after_insert` in sql alchemy to route this info via a simple airflow plugin.

**Comment by yupbank** [comment]
i'm not sure i follow, how would the plugin pass on the `after_insert` info to a DAG execution?

**Comment by MinusYuan** [comment]
> In manual jobs there is a specific user who triggered it - that doesn't exist for scheduled jobs. What value are you expecting to have for scheduled jobs?

Hi, I want to ask which variable/what kind of method can I use to get this specific user when manual trigger happens?
Now, I also need this to know which user trigger.

**Comment by avijstein** [comment]
In case folks are still looking for a solution to this (or future folks are looking, as I was last week), more recent versions of Airflow (3.1 and higher, I believe) allow you to pull `triggering_user_name` from the `dag_run` table in context. This has solved our issue of logging manual DAG triggers.

**Comment by raphaelauv** [comment]
working solution with airflow 3.3.1

```python
from airflow.sdk import DAG, get_current_context, CronDataIntervalTimetable, timezone
from airflow.providers.standard.operators.python import PythonOperator


def print_triggering_user() -> None:
    context = get_current_context()
    dag_run = context.get("dag_run")

    print(f"Run ID: {dag_run.run_id}")
    print(f"Run type: {dag_run.run_type}")

    # Populated for UI/API-triggered runs by an authenticated user.
    # None for scheduler/backfill-triggered runs.
    triggering_user = dag_run.triggering_user_name
    if triggering_user:
        print(f"Triggering user: {triggering_user}")
    else:
        print("Triggering user: N/A (not a user-triggered run)")


with DAG(
    dag_id="print_triggering_user_dag",
    schedule=CronDataIntervalTimetable("0 0 1 * *", "Europe/Paris"),
    start_date=timezone.parse("2026-07-05T00:00:00", "Europe/Paris"),
    catchup=False,
):
    print_user_task = PythonOperator(
        task_id="print_triggering_user",
        python_callable=print_triggering_user,
    )

```
